[openssh] Add openssh application for vcpkg

melchi45 opened the pull request:
- Add openssh version 9.1 with github potable application
- This application build on Linux

(cherry picked from commit b9a2f035f519dcc45da8cf6914a1de06f84ac8a6)

yhkim-77 committed:
[openssh] Add openssh application for vcpkg

- Add openssh version 9.1 with github potable application
- This application build on Linux

(cherry picked from commit b9a2f035f519dcc45da8cf6914a1de06f84ac8a6)

yhkim-77 committed:
Merge branch 'origin/master' into melchi45/master

MonicaLiu0311 commented on ports/openssh/vcpkg.json:
```diff
@@ -0,0 +1,28 @@
+  "port-version": 1,
+  "description": "OpenSSH is the premier connectivity tool for remote login with the SSH protocol.",
+  "homepage": "https://www.openssh.com",
+  "license": "BSD-3-Clause",
```
```suggestion
  "license": "BSD-3-Clause",
  "supports": "linux",
```

MonicaLiu0311 replied:
```suggestion
  "license": "BSD-3-Clause",
  "supports": "linux",
```

>This application build on Linux

If only linux is supported, please add `supports`.

MonicaLiu0311 commented on ports/openssh/portfile.cmake:
```diff
@@ -0,0 +1,49 @@
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/etc")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/libexec")
+
+file(INSTALL "${SOURCE_PATH}/LICENCE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENCE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

Please leave a blank line at the end of portfile.cmake and vcpkg.json.
(format requirement)

MonicaLiu0311 commented on ports/openssh/portfile.cmake:
```diff
@@ -0,0 +1,49 @@
+if(NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    message(FATAL_ERROR "Package only supports Linux platform.")
+endif()
+
```
Replace this part with the `support` of `vcpkg.json`

MonicaLiu0311 commented:
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).
Also please run command `vcpkg x-add-version openssh` then commit the changes.

melchi45 committed:
Update ports/openssh/vcpkg.json

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

melchi45 committed:
Update ports/openssh/portfile.cmake

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

melchi45 committed:
Update portfile.cmake

melchi45 reviewed (commented):
I was changed your suggestion.

dg0yt commented:
Does this port add a library or an application. If the latter, I would assume it won't be accepted.

melchi45 reviewed (commented):
please check again.

MonicaLiu0311 commented on ports/openssh/vcpkg.json:
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "openssh",
+  "version": "9.1",
+  "port-version": 1,
```
```suggestion
```

`port-verison` is what needs to be changed when vcpkg adjusts the port. It needs to be deleted when the port is added for the first time or the `"version"` is updated.

MonicaLiu0311 commented:
The `baseline` and `o-/openssh.json` files are missing, please refer to: [versions/](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-files-in-versions-of-any-modified-ports)

yhkim-77 committed:
[openssh] add version information

yhkim-77 committed:
Merge branch 'origin/master' into melchi45/master

melchi45 committed:
Update ports/openssh/vcpkg.json

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

melchi45 committed:
Update ports/openssh/vcpkg.json

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

melchi45 reviewed (commented):
ok, I added the version information for openssh.

MonicaLiu0311 commented:
> ok, I added the version information for openssh.

Please push the `baseline.json` file.

yhkim-77 committed:
[openssh] push the versions/baseline.json file for openssh

melchi45 reviewed (commented):
OK. I was changed the baseline.json file for openssh.

MonicaLiu0311 commented on versions/baseline.json:
```diff
@@ -5311,6 +5311,10 @@
     "oatpp-mongo": {
       "baseline": "1.3.0",
       "port-version": 1
+    },
+	"oatpp-openssh": {
```
```suggestion
    },
    "oatpp-openssh": {
```

MonicaLiu0311 replied:
Please run `./vcpkg x-add-version openssh` to generate `openssh.json` and `baseline.json`, and do not manually modify these two files.
1) The indentation format is incorrect.
2) And because of the generation problem of the `json` file, CI error was also generated：
```
##[error]Found the following errors:
    Error: While reading versions for port openssh from file: C:\a\1\s\versions\o-\openssh.json
       Version `9.1` was not found in versions file.
       Run:

           vcpkg x-add-version openssh

       to add the new port version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all
##[error]PowerShell exited with code '1'.
Finishing: Validate version files
```

MonicaLiu0311 replied:
After generating the baseline for the first time, if it is any additional modification for the same git branch, please use `./vcpkg x-add-version drogon --overwrite-version`.
(But you have a problem with the first generation, please use `./vcpkg x-add-version` drogon to complete the first generation)

melchi45 committed:
Update versions/baseline.json

Co-authored-by: MonicaLiu <110024546+MonicaLiu0311@users.noreply.github.com>

yhkim-77 committed:
[openssh] generating the baseline information for openssh

- $ vcpkg format-manifest ports/openssh/vcpkg.json
- $ vcpkg x-add-version openssh

melchi45 reviewed (commented):
[openssh] generating the baseline information for openssh

- $ vcpkg format-manifest ports/openssh/vcpkg.json
- $ vcpkg x-add-version openssh

MonicaLiu0311 commented:
Please resolve CI conflicts.
```
##[error]Found the following errors:
    Error: While reading versions for port openssh from file: C:\a\1\s\versions\o-\openssh.json
       File declares version `9.1` with SHA: 075b6bedcaebea574d5a57cbe514e3f925fc19c8
       But local port with the same version has a different SHA: a9d1547b2dfb19e05f24ddbeb1faafa4d8309a86
       Please update the port's version fields and then run:

           vcpkg x-add-version openssh
           git add versions
           git commit -m "Update version database"

       to add a new version.
To attempt to resolve all errors at once, run:
vcpkg x-add-version --all
##[error]PowerShell exited with code '1'.
Finishing: Validate version files

```

yhkim-77 committed:
[openssh] Update version database

melchi45 reviewed (commented):
Update version database

yhkim-77 committed:
[openssh] add -lpthread option for x64-linux build, but this option do not needs arm64-linux.

melchi45 reviewed (commented):
[openssh] add -lpthread option for x64-linux build, but this option do not needs arm64-linux.

yhkim-77 committed:
[openssh] change post build check script for openssh

melchi45 reviewed (commented):
change post build check script for openssh

yhkim-77 committed:
[openssh] git-tree version

melchi45 reviewed (commented):
[openssh] change git-tree version

yhkim-77 committed:
[openssh] add VCPKG_POLICY_EMPTY_INCLUDE_FOLDER option for openssh, this project is only executable binary tools to openssh.

melchi45 reviewed (commented):
[openssh] add VCPKG_POLICY_EMPTY_INCLUDE_FOLDER option for openssh, this project is only executable binary tools to openssh.

dg0yt commented:
> Does this port add a library or an application? If the latter, I would assume it won't be accepted.

Now answering my own question from the CI results:
**This port doesn't add a library.** I think it won't be accepted. (Even most build tools are rejected.)

~~~
openssh:x64-linux:/debug/libexec/sftp-server
openssh:x64-linux:/debug/libexec/ssh-keysign
openssh:x64-linux:/debug/libexec/ssh-pkcs11-helper
openssh:x64-linux:/debug/libexec/ssh-sk-helper
openssh:x64-linux:/libexec/sftp-server
openssh:x64-linux:/libexec/ssh-keysign
openssh:x64-linux:/libexec/ssh-pkcs11-helper
openssh:x64-linux:/libexec/ssh-sk-helper
openssh:x64-linux:/share/openssh/copyright
openssh:x64-linux:/share/openssh/man1/scp.1
openssh:x64-linux:/share/openssh/man1/sftp.1
openssh:x64-linux:/share/openssh/man1/ssh-add.1
openssh:x64-linux:/share/openssh/man1/ssh-agent.1
openssh:x64-linux:/share/openssh/man1/ssh-keygen.1
openssh:x64-linux:/share/openssh/man1/ssh-keyscan.1
openssh:x64-linux:/share/openssh/man1/ssh.1
openssh:x64-linux:/share/openssh/man5/moduli.5
openssh:x64-linux:/share/openssh/man5/ssh_config.5
openssh:x64-linux:/share/openssh/man5/sshd_config.5
openssh:x64-linux:/share/openssh/man8/sftp-server.8
openssh:x64-linux:/share/openssh/man8/ssh-keysign.8
openssh:x64-linux:/share/openssh/man8/ssh-pkcs11-helper.8
openssh:x64-linux:/share/openssh/man8/ssh-sk-helper.8
openssh:x64-linux:/share/openssh/man8/sshd.8
openssh:x64-linux:/share/openssh/vcpkg.spdx.json
openssh:x64-linux:/share/openssh/vcpkg_abi_info.txt
openssh:x64-linux:/tools/openssh/bin/scp
openssh:x64-linux:/tools/openssh/bin/sftp
openssh:x64-linux:/tools/openssh/bin/ssh
openssh:x64-linux:/tools/openssh/bin/ssh-add
openssh:x64-linux:/tools/openssh/bin/ssh-agent
openssh:x64-linux:/tools/openssh/bin/ssh-keygen
openssh:x64-linux:/tools/openssh/bin/ssh-keyscan
openssh:x64-linux:/tools/openssh/debug/bin/scp
openssh:x64-linux:/tools/openssh/debug/bin/sftp
openssh:x64-linux:/tools/openssh/debug/bin/ssh
openssh:x64-linux:/tools/openssh/debug/bin/ssh-add
openssh:x64-linux:/tools/openssh/debug/bin/ssh-agent
openssh:x64-linux:/tools/openssh/debug/bin/ssh-keygen
openssh:x64-linux:/tools/openssh/debug/bin/ssh-keyscan
openssh:x64-linux:/tools/openssh/debug/sbin/sshd
openssh:x64-linux:/tools/openssh/sbin/sshd
~~~

MonicaLiu0311 commented:
@melchi45 
The `port` you add must generate a `dynamic/static library` (it can also be a `herder-only` library), and vcpkg does not accept ports that **only** have `executable files`.

MonicaLiu0311 commented:
Convert this PR to draft since there is no progress. Please ping us if this PR is ready for review again.

MonicaLiu0311 commented:
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.
